### PR TITLE
Tpbot/Bot presentation in service

### PR DESCRIPTION
**Pull request by Mecharyry**
#### Problem
The `Bot` is able to automatically open a `hangout` when connected to the middleman server. When opening the `hangout` the `BotActivity` is torn down including any necessary presenters. Tearing down the presentation layer of the `Bot` automatically disconnects said `Bot` from the middleman server. 

#### Solution
Create a `BoundService` and perform all `Bot` actions in this service. For now, only the logic related to connecting and listening for moves has been moved. In the next PR the movement service will be added to the `BoundService` rather than having two independent services (if possible).
The unbinding of the `BotService` is performed in `onDestroy` ensuring that the service lives even when the app is moved to the background (similar how multi-window is handled). 

#### Screen Capture
No updates.

### CODE REVIEW

File: TelepresenceBot/mobile/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -60,6 +61,10 @@
       android:name="com.novoda.tpbot.bot.MovementService"
       android:exported="false" />
 
+    <service
+      android:name="com.novoda.tpbot.bot.BotService"
+      android:exported="false" />
+
```
Comment (jackSzm) [inline TelepresenceBot/mobile/src/main/AndroidManifest.xml]: 👍

File: TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/bot/BotActivity.java [hunk 2]
```diff
@@ -174,10 +174,15 @@ protected void onStop() {
             unbindService(serviceConnection);
             boundToMovementService = false;
         }
-        presenter.stopPresenting();
         super.onStop();
     }
 
+    @Override
+    protected void onDestroy() {
```
Comment (jackSzm) [inline TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/bot/BotActivity.java]: So basically we would still close the connection when activity is destroyed ?

So having this is mind when device would low run in memory and decide to kill the activities that are in the background , then the service would be destroyed as well right ?

Comment (Mecharyry) [inline TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/bot/BotActivity.java]: Yes exactly. We want it to be available as long as the activity is bound in the activity backstack. If the phone runs out of memory / the activity is killed by the user then everything is dropped, including the service.

Comment (jackSzm) [inline TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/bot/BotActivity.java]: LGTM then !

**Pull request merged**